webapp/src/main/webapp/src/app/app.component.ts

```diff
@@ -21,26 +22,30 @@ export class AppComponent {
   /*
   Even though the angulartics2GoogleAnalytics variable is not explicitly used, without it analytics data is not sent to the service
    */
-  constructor(public translate: TranslateService, private _userService: UserService,
+  constructor(public translate: TranslateService,
+              private _userService: UserService,
+              private router: Router,
               private angulartics2GoogleAnalytics: Angulartics2GoogleAnalytics) {
 
-    let languages = [ 'en', 'ja' ];
-    let defaultLanguage = languages[ 0 ];
-    translate.addLangs(languages);
+    let defaultLanguage = SupportedLanguages[ 0 ];
+    translate.addLangs(SupportedLanguages);
     translate.setDefaultLang(defaultLanguage);
-    translate.use(languages.indexOf(translate.getBrowserLang()) != -1 ? translate.getBrowserLang() : defaultLanguage);
+    translate.use(SupportedLanguages.indexOf(translate.getBrowserLang()) != -1 ? translate.getBrowserLang() : defaultLanguage);
 
   }
 
   ngOnInit() {
     this._userService.getCurrentUser().subscribe(user => {
 
       this._user = user;
+      let userHasAccess = user.permissions.some(permission => RequiredPermissions.indexOf(permission) !== -1);
 
-      this.userHasAccess = user.permissions.some(permission => permission == "GROUP_WRITE");
+      if (!userHasAccess) {
+        this.router.navigate([ 'access-denied' ]);
+      }
```

Comment:
Now here in the AppComponent ngInit if we find the user has none of the RequiredPermissions then we forward them to access-denied.

Reply:
I feel like there would be a more angular way to do this than in ngOnInit. Can the "canActivate" route hooks not redirect you based on absence of permissions?

Reply:
I did something like that more along those lines in Reporting since we had a home component and a home route.  Here, we don't really have that and start navigation on a functional page.  Until we have some root home route I don't think it makes sense to do that now.

And as for doing it on AppComponent, you can't because it's the root and is not bound to a ui route.

Reply:
Hm. Wonder if there was a better pattern we could have followed where the app component just acted as an empty shell and the whole app view is delivered through the routes.